app/lib/registries/full_topic_taxonomy_registry.rb

```diff
@@ -6,6 +6,21 @@ def [](base_path)
       taxonomy[base_path]
     end
 
+    def find_parent_by_id id
+      topic = taxonomy.select { |_, value|
+        value['content_id'] == id
+      }
+
+      path = topic.keys[0]
+
+      title = topic.dig(path, 'title')
+
+      {
+        path: path,
+        title: title,
+      }
+    end
```

Comment:
This returns `{:path=>nil, :title=>nil}` if the id isn't found, is that correct?

Reply:
This looks good. I'd recommend `find` since that's guaranteed to return only one item, whereas [`select`](https://apidock.com/ruby/Hash/select) could return multiple.

Maybe something like this?

```
path, topic = taxonomy.find { |_, value| value['content_id'] == id } || nil, {}
title = topic.dig(path, 'title')
{ path: path, title: title }
```

Reply:
Good shout on both points. I've used the find suggested and fixed the method to return `nil` if there's nothing, otherwise it returns the hash - this also makes the check for the topic more robust.